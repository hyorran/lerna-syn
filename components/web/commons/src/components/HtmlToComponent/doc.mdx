---
name: HtmlToComponent
menu: Commons
---

import { Playground, PropsTable } from 'docz'
import HtmlToComponent, { ComponentWithProps } from './'

# Helper Page

A Simple HelperPage Example 

## Basic Usage
<Playground>
  <HtmlToComponent html="<h2>My html as component</h2>" />
</Playground>

## Props

<PropsTable of={ ComponentWithProps } />
