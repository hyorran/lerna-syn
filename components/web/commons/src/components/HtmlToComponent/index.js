import React from 'react'
import PropTypes from 'prop-types'
import HtmlToReactParser from 'html-to-react/lib/parser'

const HtmlToComponent = (props) => {
  const { parse } = new HtmlToReactParser()
  return (
    <div className={ props.containerClass }>
      { parse(props.html) }
    </div>
  )
}

HtmlToComponent.propTypes = {
  /** Transform an HTML into a React.Component */
  html: PropTypes.string.isRequired,
  /** Component's material css class. */
  containerClass: PropTypes.string
}

HtmlToComponent.defaultProps = {
  containerClass: null
}

export default HtmlToComponent
export { HtmlToComponent as ComponentWithProps }
