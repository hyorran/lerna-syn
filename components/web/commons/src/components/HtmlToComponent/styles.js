export default () => ({
  containerClass: {
    textAlign: 'justify'
  }
})
