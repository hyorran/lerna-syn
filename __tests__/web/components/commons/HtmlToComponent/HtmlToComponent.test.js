import React from 'react'
import HtmlToComponent from '@syntesis/c-commons/src/components/HtmlToComponent'
import { mount } from 'enzyme'

describe('<HtmlToComponent />', () => {
  it('rendered', () => {
    const wrapper = mount((
      <HtmlToComponent html=" <h2>Html as Component </h2> " />
    ))
    expect(wrapper).toMatchSnapshot()
  })
})
