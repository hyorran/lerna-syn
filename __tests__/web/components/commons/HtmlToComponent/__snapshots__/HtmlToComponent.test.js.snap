// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HtmlToComponent /> rendered 1`] = `
<HtmlToComponent
  containerClass={null}
  html=" <h2>Html as Component </h2> "
>
  <div
    className={null}
  >
     
    <h2
      key="1"
    >
      Html as Component 
    </h2>
     
  </div>
</HtmlToComponent>
`;
